camp/accounts/views.py

```diff
@@ -9,3 +18,66 @@ class ProfileDetailView(LoginRequiredMixin, DetailView):
 
     def get_object(self):
         return self.request.user
+
+
+class MembershipListView(AutoPermissionRequiredMixin, ListView):
+    permission_type_map = [
+        (CreateView, "add"),
+        (UpdateView, "change"),
+        (DeleteView, "delete"),
+        (DetailView, "view"),
+        (ListView, "view"),
+    ]
+
+    model = Membership
+    template_name = "account/membership_list.html"
+    context_object_name = "membership_list"
+
+    def get_queryset(self):
+        queryset = Membership.objects.all()
+        return queryset.filter(user=self.request.user)
+
+
+class MembershipDetailView(AutoPermissionRequiredMixin, DetailView):
+    model = Membership
+    template_name = "account/membership_detail.html"
+
+
+class MembershipCreateView(AutoPermissionRequiredMixin, CreateView):
+    model = Membership
+    fields = ["nickname"]
+    success_url = reverse_lazy("membership-list")
+    template_name = "account/membership_create.html"
+
+    def form_valid(self, form):
+        # You don't get to choose which Game the Membership applies to.
+        # Intercept the model to set the Game manually before saving.
+        self.object = form.save(commit=False)
+        self.object.user = self.request.user
+        self.object.game = self.request.Game
+        try:
+            self.object.save()
+        except IntegrityError:
+            form.add_error("user", "User already has a membership for this game.")
+            return super().form_invalid(form)
+        return super().form_valid(form)
+
+
+class MembershipUpdateView(AutoPermissionRequiredMixin, UpdateView):
+    model = Membership
+    fields = ["nickname"]
+    success_url = reverse_lazy("membership-list")
+    template_name = "account/membership_update.html"
+
+    def form_valid(self, form):
```

Comment:
This probably shouldn't be on the Update view.
Consider what happens if an admin tries to update another user's membership details.